### PR TITLE
ICS26: updated upgrade callback signatures

### PR DESCRIPTION
**Pull request by crodriguezvega**
closes: #1012

### CODE REVIEW

File: spec/core/ics-026-routing-module/UPGRADES.md [hunk 1]
```diff
@@ -26,11 +26,11 @@ In order to support channel upgrades, the application must implement the followi
...
 interface ModuleUpgradeCallbacks {
-    onChanUpgradeInit: onChanUpgradeInit,
-    onChanUpgradeTry: onChanUpgradeTry,
-    onChanUpgradeAck: onChanUpgradeAck,
-    onChanUpgradeConfirm: onChanUpgradeConfirm,
-    onChanUpgradeRestore: onChanUpgradeRestore
+  onChanUpgradeInit: onChanUpgradeInit,
+  onChanUpgradeTry: onChanUpgradeTry,
+  onChanUpgradeAck: onChanUpgradeAck,
+  onChanUpgradeConfirm: onChanUpgradeConfirm,
```
Comment (DimitrisJim) [inline spec/core/ics-026-routing-module/UPGRADES.md:32]:
```suggestion
  onChanUpgradeOpen: onChanUpgradeOpen,
```

File: spec/core/ics-026-routing-module/UPGRADES.md [hunk 2]
```diff
@@ -48,14 +48,12 @@ If an error is returned, then core IBC will revert any changes made by `onChanUp
...
 function onChanUpgradeInit(
-  order: ChannelOrder,
-  connectionHops: [Identifier],
   portIdentifier: Identifier,
   channelIdentifier: Identifier,
-  counterpartyPortIdentifier: Identifier,
-  counterpartyChannelIdentifier: Identifier,
-  version: string,
-  previousVersion: string) => (version: string, err: Error) {
+  upgradeSequence: uint64,
```
Comment (DimitrisJim) [inline spec/core/ics-026-routing-module/UPGRADES.md:53]: ah, spec for ics-004 has this after the proposed upgrade field args, probably should be as you have it here tho

File: spec/core/ics-026-routing-module/UPGRADES.md [hunk 3]
```diff
@@ -105,23 +101,22 @@ If the callback returns successfully, the application MUST have its state fully
 function onChanUpgradeAck(
   portIdentifier: Identifier,
   channelIdentifier: Identifier,
-  counterpartyChannelIdentifier: Identifier, 
-  counterpartyVersion: string) {
+  counterpartyVersion: string) => Error {
```
Comment (DimitrisJim) [inline spec/core/ics-026-routing-module/UPGRADES.md:104]: `counterpartyUpgradeVersion` maybe?

**Review by DimitrisJim — APPROVED** [review]
minor nits, other than that, lgtm!

**Comment by DimitrisJim** [comment]
tho might be best to way for finalizing sigs after implementation as pointed out in https://github.com/cosmos/ibc-go/pull/4478#discussion_r1308436933

File: spec/core/ics-026-routing-module/UPGRADES.md (hunk 1 above)
Comment (AdityaSripal) [reply spec/core/ics-026-routing-module/UPGRADES.md:32]: Agreed with this since it will be called on both sides

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**